### PR TITLE
bpo-15718: Document the upper bound constrain on __len__ return value.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -2011,6 +2011,14 @@ through the container; for mappings, :meth:`__iter__` should be the same as
    :meth:`__bool__` method and whose :meth:`__len__` method returns zero is
    considered to be false in a Boolean context.
 
+.. impl-detail::
+
+   In CPython, the length is required to be at most :attr:`sys.maxsize`.
+   If the length is larger than ``sys.maxsize`` some features (such as
+   :func:`len`) may raise :exc:`OverflowError`.  To prevent raising
+   ``OverflowError`` by truth value testing an object must define a
```
Comment (louisom) [inline Doc/reference/datamodel.rst]: small nit point, add a comma after truth value testing

`by truth value testing, an object must ...`

**Review by louisom — COMMENTED** [review]
Small nit point, other LGTM

**Review by terryjreedy — APPROVED** [review]
With comma

File: Doc/reference/datamodel.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/reference/datamodel.rst]: Thanks @lulouie!

**Pull request merged**